src/core/lib/iomgr/tcp_posix.c

```diff
@@ -107,7 +112,16 @@ static void tcp_handle_write(grpc_exec_ctx *exec_ctx, void *arg /* grpc_tcp */,
 
 static void tcp_shutdown(grpc_exec_ctx *exec_ctx, grpc_endpoint *ep) {
   grpc_tcp *tcp = (grpc_tcp *)ep;
-  grpc_fd_shutdown(exec_ctx, tcp->em_fd);
+  gpr_mu_lock(&tcp->mu);
```

Comment:
My understanding was that tcp_shutdown was designed to be called asynchronously. This PR only adds one more place from where tcp_shutdown might is called. The behavior after shutdown is unchanged.

In any case, a quick reading of the code suggests that reading/writing on already shutdown FDs will generate errors that are handled appropriately.